pvlib/bifacial/utils.py

```diff
@@ -0,0 +1,151 @@
+        Surface tilt angle. The tilt angle is defined as
+        degrees from horizontal, e.g., surface facing up = 0, surface facing
+        horizon = 90. [degree]
+    surface_azimuth: numeric
+        Azimuth of the module surface, i.e., North=0, East=90, South=180,
+        West=270. [degree]
+    solar_zenith : numeric
+        Solar zenith angle. [degree].
+    solar_azimuth : numeric
+        Solar azimuth. [degree].
+    gcr : float
+        Ground coverage ratio, which is the ratio of row slant length to row
+        spacing (pitch). [unitless]
+    max_zenith : numeric, default 87
+        Maximum zenith angle. For solar_zenith > max_zenith, unshaded ground
+        fraction is set to 0. [degree]
+
+    Returns
+    -------
+    f_gnd_beam : numeric
+        Fraction of distance betwen rows (pitch) with direct irradiance
+        (unshaded). [unitless]
+
+    References
+    ----------
+    .. [1] Mikofksi, M., Darawali, R., Hamer, M., Neubert, A., and Newmiller,
+       J. "Bifacial Performance Modeling in Large Arrays". 2019 IEEE 46th
+       Photovoltaic Specialists Conference (PVSC), 2019, pp. 1282-1287.
+       doi: 10.1109/PVSC40753.2019.8980572.
+    """
+    tan_phi = _solar_projection_tangent(solar_zenith, solar_azimuth,
+                                        surface_azimuth)
+    f_gnd_beam = 1.0 - np.minimum(
+        1.0, gcr * np.abs(cosd(surface_tilt) + sind(surface_tilt) * tan_phi))
+    np.where(solar_zenith > max_zenith, 0., f_gnd_beam)  # [1], Eq. 4
+    return f_gnd_beam  # 1 - min(1, abs()) < 1 always
+
+
+def _vf_ground_sky_2d(x, rotation, gcr, pitch, height, max_rows=10):
+    r"""
+    Calculate the fraction of the sky dome visible from point x on the ground.
+
+    The view factor accounts for the obstruction of the sky by array rows that
+    are assumed to be infinitely long.  View factors are thus calculated in
+    a 2D geometry. The ground is assumed to be flat and level.
+
+    Parameters
+    ----------
+    x : numeric
+        Position on the ground between two rows, as a fraction of the pitch.
+        x = 0 corresponds to the point on the ground directly below the
+        center point of a row. Positive x is towards the right. [unitless]
+    rotation : float
+        Rotation angle of the row's left edge relative to row center. [degree]
+    gcr : float
+        Ratio of the row slant length to the row spacing (pitch). [unitless]
+    height : float
+        Height of the center point of the row above the ground; must be in the
+        same units as ``pitch``.
+    pitch : float
+        Distance between two rows; must be in the same units as ``height``.
+    max_rows : int, default 10
+        Maximum number of rows to consider on either side of the current
+        row. [unitless]
+
+    Returns
+    -------
+    vf : numeric
+        Fraction of sky dome visible from each point on the ground. [unitless]
+    wedge_angles : array
+        Angles defining each wedge of visible sky. Shape is
```

Comment:
I think as it currently stands this return value is actually the bounding angles for each row, not the bounding angles for each visible sky wedge.  No preference from me on whether the docstring is changed to match the code or vice versa.